### DISCUSSION TITLE
Rename `tea --which`

### DISCUSSION THREAD

**Discussion by davdroman** [pkgxdev/pkgx#702] [Ideas]
My expectation of running `tea --which` is that it behaves as the UNIX `which` command, which outputs the path of whichever command is passed in. However this is not currently the case:

```sh
$ which node
/usr/local/bin/node

$ tea --which node
nodejs.org
```

Instead, I'd expect something like:

```sh
$ tea --which node
/Users/davdroman/.tea/nodejs.org/v20.6.1/bin/node

$ tea --which node@14
/Users/davdroman/.tea/nodejs.org/v14.21.3/bin/node
```

Now granted, `tea` is supposed to hide bin paths as much as possible and handle the resolving for you, so maybe this `--which` flag isn't even something useful to have. However, it's still confusing and sets the wrong expectation for the user.

Maybe just renaming the flag is the way to go? Here are a few ideas:

```sh
$ tea --full node
nodejs.org

$ tea --fullname node
nodejs.org

$ tea --origin node
nodejs.org

$ tea --pantry node
nodejs.org
```

**Comment by mxcl** [comment]
Fair point. This is used by the shellcode to determine if we provide a program or not. Thus it returns the fully qualified name.

`--has` perhaps?

**Reply by davdroman** [reply]
Love `--has` 👌

**Reply by mxcl** [reply]
K will add to `alpha.2` thanks for the input! Was a good point.

**Reply by mxcl** [reply]
Sorry didn't get it into `alpha.2`. I spent some time thinking and was trying to come up with something more suitable. It's a very specific flag as it stands.

**Reply by mxcl** [reply]
`alpha.4` renamed this `--provider`
